pkg/services/grafana-apiserver/storage/entity/utils_test.go

```diff
@@ -0,0 +1,211 @@
+package entity
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana/pkg/apis/playlist/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/grn"
+	entityStore "github.com/grafana/grafana/pkg/services/store/entity"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+func TestResourceToEntity(t *testing.T) {
+	createdAt := metav1.Now()
+	createdAtStr := createdAt.UTC().Format(time.RFC3339)
+
+	updatedAt := createdAt.Add(time.Hour)
+	updatedAtStr := updatedAt.UTC().Format(time.RFC3339)
+
+	apiVersion := "v0alpha1"
+	requestInfo := &request.RequestInfo{
+		APIVersion: apiVersion,
+	}
+
+	testCases := []struct {
+		key                        string
+		resource                   runtime.Object
+		expectedKey                string
+		expectedTenantID           int64
+		expectedResourceKind       string
+		expectedResourceIdentifier string
+		expectedResourceGroup      string
+		expectedGroupVersion       string
+		expectedName               string
+		expectedGuid               string
+		expectedVersion            string
+		expectedFolder             string
+		expectedCreatedAt          int64
+		expectedUpdatedAt          int64
+		expectedCreatedBy          string
+		expectedUpdatedBy          string
+		expectedSlug               string
+		expectedOrigin             *entityStore.EntityOriginInfo
+		expectedLabels             map[string]string
+		expectedMeta               []byte
+		expectedBody               []byte
+	}{
+		{
+			key: "/playlist.grafana.app/playlists/default/test-uid",
+			resource: &v0alpha1.Playlist{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: apiVersion,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					CreationTimestamp: createdAt,
+					Labels:            map[string]string{"label1": "value1", "label2": "value2"},
+					Name:              "test-name",
+					ResourceVersion:   "1",
+					UID:               "test-uid",
+					Annotations: map[string]string{
+						"grafana.app/createdBy":        "test-created-by",
+						"grafana.app/updatedBy":        "test-updated-by",
+						"grafana.app/updatedTimestamp": updatedAtStr,
```

Comment:
what is supplied value is not represented in RFC3339 format? currently, I think it remains unconverted in the response

Reply:
good call-out, right now the [GetUpdatedTimestamp](https://github.com/grafana/grafana/blob/6659ed8722278339531634236a01bbd6b0e045ac/pkg/kinds/general.go#L237) meta accessor method will just return `nil` if it can't parse the label value.  We could update that method to support returning an error in that case, which would give us the opportunity to return an error to the user instead of just ignoring the value.

Reply:
we added errors in main to handle this situation.